addons/mail/models/mail_thread.py

```diff
@@ -613,13 +613,14 @@ def _message_track(self, tracked_fields, initial):
         for col_name, col_info in tracked_fields.items():
             initial_value = initial[col_name]
             new_value = self[col_name]
+            company_id = col_info.get('company_dependent') and self.env.company.id or False
```

Comment:
Can you write a quick test? 
- Write on company dependent field
- Try to read with current user
- Change company and try to read again (with same user if you change the search filter ass explained earlier)

Reply:
Test case is done.